src/app/Services/UserListService/user-list.service.ts

```diff
@@ -25,12 +32,21 @@ export class UserListService {
 
   constructor(private http: HttpClient, private paginationService: PaginationService) { }
 
-  getUserList() {
-    const typeUrl = this.url + 'userlist/getparams';
+  getUserList(filter?: UserListFilter, event?: PageEvent) {
+    let typeUrl = this.url + 'userlist/getparams';
+
+    if (filter != null) {
+        if (filter.searchKey !== undefined && filter.searchKey !== '') {
+          typeUrl += `?${SEARCH_NAME}=${filter.searchKey}&`;
+        }
+        if (filter.selectedStatus !== 0 && filter.selectedStatus !== undefined) {
+          typeUrl += `?${SEARCH_STATUS}=${filter.selectedStatus}`;
+        }
+    }
     return this.http.get(typeUrl, this.httpOptions);
   }
-  getUserListParams(filter: UserListFilter) {
-    const typeUrl = this.url + 'userlist/getparams';
+  getStatuses() {
+    const typeUrl = this.url + 'userlist/getstatuses';
     return this.http.get(typeUrl, this.httpOptions);
   }
```

Comment:
This service works only with userlist controller, add "userlist/" to main url and add only name of action.